modules/targetAcquisition/taxi/Taxi.py

```diff
@@ -0,0 +1,331 @@
+
+config = ConfigProto()
+config.gpu_options.allow_growth = True
+session = InteractiveSession(config=config)
+
+
+class Taxi:
+    """
+    Performs cardboard box detection on a given video frame
+
+    Attributes
+    ----------
+    state : string
+        state of object recognition ("BOX", "TRACK", ...)
+    bbox : list<tuple<tuple, tuple>>
+        a list of ((x1, y1), (x2, y2)) coordinates for (top left, bottom right) of bounding boxes; one per box
+    frame : np.ndarray
+        the current video frame
+    yolo : Detection object
+        the YOLOv5 detector
+    tracker : TrackerKCF object
+        the cv2 KCF bounding box tracker
+    nextUncheckedID : int
+        the ID of the next box to scan
+    expectedCount : int
+        the number of boxes it needs to detect to go into tracking state
+    numStableFrames : int
+        the number of consecutive frames with all <expectedCount> number of boxes in view before switching to TRACK
+    distanceFromBox : int
+        the plane's current distance from the target box
+    minDistanceFromBox : int
+        the minimum distance the plane can be from the box until it's considered too close
+    moveWaitTarget : int
+        the number of frames to wait before issuing another move or turn command
+    frameCount : int
+        the number of consecutive frames where all <expectedCount> boxes are in view
+    totalWait : int
+        maximum number of frames to wait for before time-out while searching for <expectedCount> number of boxes
+    moveWaitCount : int
+        wait for a certain number of detections before issuing the next movement command
+    distanceCount : int
+        if the plane has moved by <distanceCount> number of steps forward, it needs to recalibrate the bounding box by calculating a new target distance
+        
+    Constants
+    ----------
+    FOCAL_LENGTH : int
+        focal length of the camera (in mm)
+    REAL_HEIGHT : int
+        real height of the box (in mm)
+    IMAGE_HEIGHT : int
+        height of the image (in pixels)
+    SENSOR_HEIGHT : int
+        height of the sensor (in mm)
+   
+    Methods
+    -------
+    __init__()
+        Initialize class variables
+    find_overlapped_bbox()
+        Find which new YOLO bbox overlaps the previously tracked bbox
+    calculate_distance()
+        Calculate approximate distance between box and drone
+    set_state(state: string)
+        Prepare variables for box detection, etc.
+    main()
+        Main operations: getting camera input and passing the image to appropriate methods
+    """
+    
+    # Constant declaration
+    
+    FOCAL_LENGTH = 24
+    REAL_HEIGHT = 101.6
+    IMAGE_HEIGHT = 1080
+    SENSOR_HEIGHT = 4.63
+    
+    def __init__(self, state="BOX", bbox=[((0, 0), (0, 0))], frame=[], nextUncheckedID=0,
+                 expectedCount=5, numStableFrames=20, distanceFromBox=0,
+                 minDistanceFromBox=0, moveWaitTarget=0, recalibrate=False, lastBbox=[],
```

Comment:
Should set some of these values (particularly minDistanceFromBox) to something nonzero